### ISSUE TITLE
Regression in apache_module.enable test mode on Salt 2015.5.3

### ISSUE THREAD

**Issue by HontoNoRoger** [saltstack/salt#26454]
Hello, I'm getting an exception when using `state.highstate test=True` which has to enable Apache mods via `apache_module.enable`. This regression was introduced in 2015.5.3 it seems. I hadn't had the problem in 2015.5.2 I guess.

Normal mode (non-test mode) works fine.

```
# salt-call state.highstate test=True

…

[INFO    ] {'newfile': '/etc/apache2/mods-available/proxy.conf'}
[INFO    ] Completed state [/etc/apache2/mods-available/proxy.conf] at time 13:03:06.137450
[ERROR   ] An un-handled exception was caught by salt's global exception handler:
KeyError: 'apache_module.enable'
Traceback (most recent call last):
  File "/usr/bin/salt-call", line 11, in <module>
    salt_call()
  File "/usr/lib/python2.7/dist-packages/salt/scripts.py", line 227, in salt_call
    client.run()
  File "/usr/lib/python2.7/dist-packages/salt/cli/call.py", line 69, in run
    caller.run()
  File "/usr/lib/python2.7/dist-packages/salt/cli/caller.py", line 236, in run
    ret = self.call()
  File "/usr/lib/python2.7/dist-packages/salt/cli/caller.py", line 138, in call
    ret['return'] = func(*args, **kwargs)
  File "/usr/lib/python2.7/dist-packages/salt/modules/state.py", line 523, in highstate
    whitelist=kwargs.get('whitelist')
  File "/usr/lib/python2.7/dist-packages/salt/state.py", line 2968, in call_highstate
    return self.state.call_high(high)
  File "/usr/lib/python2.7/dist-packages/salt/state.py", line 2067, in call_high
    ret = dict(list(disabled.items()) + list(self.call_chunks(chunks).items()))
  File "/usr/lib/python2.7/dist-packages/salt/state.py", line 1623, in call_chunks
    running = self.call_chunk(low, running, chunks)
  File "/usr/lib/python2.7/dist-packages/salt/state.py", line 1858, in call_chunk
    running = self.call_chunk(chunk, running, chunks)
  File "/usr/lib/python2.7/dist-packages/salt/state.py", line 1769, in call_chunk
    self._mod_init(low)
  File "/usr/lib/python2.7/dist-packages/salt/state.py", line 612, in _mod_init
    self.states['{0}.{1}'.format(low['state'], low['fun'])]  # pylint: disable=W0106
  File "/usr/lib/python2.7/dist-packages/salt/utils/lazy.py", line 90, in __getitem__
    raise KeyError(key)
KeyError: 'apache_module.enable'
Traceback (most recent call last):
  File "/usr/bin/salt-call", line 11, in <module>
    salt_call()
  File "/usr/lib/python2.7/dist-packages/salt/scripts.py", line 227, in salt_call
    client.run()
  File "/usr/lib/python2.7/dist-packages/salt/cli/call.py", line 69, in run
    caller.run()
  File "/usr/lib/python2.7/dist-packages/salt/cli/caller.py", line 236, in run
    ret = self.call()
  File "/usr/lib/python2.7/dist-packages/salt/cli/caller.py", line 138, in call
    ret['return'] = func(*args, **kwargs)
  File "/usr/lib/python2.7/dist-packages/salt/modules/state.py", line 523, in highstate
    whitelist=kwargs.get('whitelist')
  File "/usr/lib/python2.7/dist-packages/salt/state.py", line 2968, in call_highstate
    return self.state.call_high(high)
  File "/usr/lib/python2.7/dist-packages/salt/state.py", line 2067, in call_high
    ret = dict(list(disabled.items()) + list(self.call_chunks(chunks).items()))
  File "/usr/lib/python2.7/dist-packages/salt/state.py", line 1623, in call_chunks
    running = self.call_chunk(low, running, chunks)
  File "/usr/lib/python2.7/dist-packages/salt/state.py", line 1858, in call_chunk
    running = self.call_chunk(chunk, running, chunks)
  File "/usr/lib/python2.7/dist-packages/salt/state.py", line 1769, in call_chunk
    self._mod_init(low)
  File "/usr/lib/python2.7/dist-packages/salt/state.py", line 612, in _mod_init
    self.states['{0}.{1}'.format(low['state'], low['fun'])]  # pylint: disable=W0106
  File "/usr/lib/python2.7/dist-packages/salt/utils/lazy.py", line 90, in __getitem__
    raise KeyError(key)
KeyError: 'apache_module.enable'
```

```
# salt-minion --versions-report
                  Salt: 2015.5.3
                Python: 2.7.6 (default, Jun 22 2015, 17:58:13)
                Jinja2: 2.7.2
              M2Crypto: 0.21.1
        msgpack-python: 0.3.0
          msgpack-pure: Not Installed
              pycrypto: 2.6.1
               libnacl: Not Installed
                PyYAML: 3.10
                 ioflo: Not Installed
                 PyZMQ: 14.0.1
                  RAET: Not Installed
                   ZMQ: 4.0.4
                  Mako: Not Installed
               Tornado: Not Installed
 Debian source package: 2015.5.3+ds-1trusty1
```

**Comment by jfindlay** [comment]
@HontoNoRoger, thanks for the report.

**Comment by gfa** [comment]
+1

**Comment by ChoOo7** [comment]
+1

**Comment by aronneagu** [comment]
+1

**Comment by rallytime** [comment]
@HontoNoRoger (and everyone else with +1s) - This was a regression in 2015.5.3. It is fixed in both 2015.5.6 and 2015.8.0. I'd be awesome if you or any others can confirm that this is fixed in either of those releases or at the HEAD of the 2015.5 branch? If so, we can call this one good to go!

The backstory here is that this bug was fixed on the 2015.8 brach with #27681, which I back-ported to 2015.5 with #27681. This stacktrace was also reported in #27619, #26882, and #26280.

**Comment by rallytime** [comment]
Update: It appears that I mis-remembered when my back-porting PR went into the 2015.5 branch and it missed the 2015.5.6 release tag. However, the initial stacktrace reported by @HontoNoRoger should be fixed at the HEAD of the 2015.5 branch as well as 2015.8.0.

**Comment by rallytime** [comment]
I am closing this bug since it is fixed on the 2015.5 branch and will be available in the 2015.5.7 release (and is also fixed in 2015.8.0 and newer). If this comes up again, please let us know and we will take another look!
